book/1-pre-project-phase/key_areas_in_data.md

```diff
@@ -0,0 +1,75 @@
+
+In practice, there are many considerations and checks for data quality. In this article, we focus on 4 key questions that are highly relevant at the pre-project phase to ensure that the AI problem is feasible and well-scoped. Note that while this is written with a supervised learning approach in mind, most of the considerations are also applicable to other learning approaches.
+
+1. Is the target clearly defined and labels available?
+
+Frequently, sponsor may not be able to articulate the target variable of interest. You will need to explain it as a feature of the dataset that the sponsor is keen to predict. This target variable is related to a business outcome (e.g. reducing employee's repetitive tasks, prioritising patients for treatment based on mortality rate) that the sponsor wants to achieve. 
+
+If the target is of classification nature, it will be ideal to observe the current label proportion (e.g. binary label: "yes", "no") to ascertain if there is presence of label data imbalance. If the label data is highly imbalanced (e.g. 5% positive label and 95% negative label), an alternative would be to reframe the AI problem as an anomaly detection problem. More importantly, you would be keen to confirm if the label data corresponds to the definition of the target and the availability of the label data. 
+
+If the target belongs to a regression type, the question is more about ensuring the target variable is truly numeric/continuous. If upon checking you find that there is only a limited set of discrete values that the target variable can take on, it is better to reframe it as a classification problem. This is a possible scenario in businesses that rely on domain experts to run product experiments by testing different combinations of inputs with varying levels.
+
+![Discreteplot](../assets/images/charts/discreteplot_chart.png)  
+
+2. Does the data include features that can predict the target?
+
+An ideal way to estimate if the relevant features are present would be to ask the sponsor the following questions based on a priori domain knowledge: 
+
+- If a human were to perform the task, what information would the person rely on to make the prediction? 
+- Is this information available for training the model?
+
+There should be adequate signal and variance within the relevant features. Signal refers to the presence of important information relevant to predicting the target. This can be checked by creating a correlation plot (example chart below) between numerical features and the target variable where the relevant features have a correlation value that is not (or close to) 0. Variance refers to having a sufficient spread of values (eg. straight line and near-constant features are useless for prediction).
```

Comment:
Replace this para with this, which is more generalisable:
+++++
There should be adequate signal and variance within the relevant features. Signal refers to the presence of sufficient information relevant to predicting the target. Variance refers to having sufficient range and diversity of data. 

This can be checked via a small, representative sample dataset from the sponsor.
- for CV and NLP problems, using a pretrained model to run inference on sample data is a great way to assess the 'information-ness'. Sometimes, you can also 'eyeball' the data, since for images and text, WYSIWYG!
- for tabular problems, you can create a plot of mutual information/correlation between features and the target variable. A simple example is shown below for numeric features.
+++++